File: ABC/ABC128/D.cpp
#include <bits/stdc++.h>
#include <atcoder/all>
using namespace std;
using namespace atcoder;

#define fs(n) fixed << setprecision(n)
#define mp(a, b) make_pair(a, b)
#define all(x) x.begin(), x.end()
#define const constexpr
#define pdesc(t) t, vector<t>, greater<t>
#define chmin(a, b) a = min(a, b)
using ll = long long;
using ld = long double;

int main(){
  int n, k; cin >> n >> k;
  vector<int> v(n);
  int ans = 0;
  for(int i = 0; i < n; i++) cin >> v[i];
  for(int left = 0; left <= k; left++){
    for(int right = 0; right <= k - left; right++){
      int down = k - left - right;
      vector<int> dq;
      int res = 0;
      for(int i = 0; i < left; i++){
        res += v[i];
        if(v[i] < 0) dq.push_back(v[i]);
      }
      for(int i = 0; i < right; i++){
        res += v[n - 1 - i];
        if(v[n - 1 - i] < 0) dq.push_back(v[n - 1 - i]);
      }
      sort(all(dq));
      for(int i = 0; i < min(down, (int)dq.size()); i++){
        res -= dq[i];
      }
      ans = max(ans, res);
    }
  }
  cout << ans << endl;
}

File: ABC/ABC128/D.py
n, k = map(int, input().split())
v: list[int] = list(map(int, input().split()))
ans = 0
for l in range(min(k + 1, n)):
    for r in range(0, min(k, n) - l + 1):
        down: int = k - l - r
        dq: list[int] = []
        res: int = 0
        for i in range(l):
            res += v[i]
            if v[i] < 0:
                dq.append(v[i])
        for i in range(1, r + 1):
            res += v[-i]
            if v[-i] < 0:
                dq.append(v[-i])
        dq = sorted(dq)
        for i in range(min(down, len(dq))):
            res -= dq[i]
        ans = max(ans, res)
print(ans)
